src/components/pagination.tsx

```diff
@@ -47,7 +48,32 @@ export const Pagination: StatelessComponent<Props> = ({
   query,
   pageLinkBuilder,
   renderHead,
+  onButtonClick,
 }) => {
+  const [currentPage, setCurrentPage] = useState(null)
+  const [navButtonClicked, setNavButtonClicked] = useState("")
+
+  const handlePageChange = (data) => {
+    setCurrentPage(data)
+  }
+
+  const handleNavButtonClick = (label) => {
+    setNavButtonClicked(label)
+  }
+
+  useEffect(() => {
```

Comment:
Do we really need this useEffect? Cant we just call the functions inside `handlePageChange` and `handleNavButtonClick`?

Reply:
Yea we actually needed. It's a bit of a complex case becasue we have a way to tell if `next/prev` buttons have been clicked, but not if pagination buttons (numbers) have been clicked.
Basically, when `next/prev` is called, we just update a state with that label. That same click will also trigger the value (page) state to change. That's why the only way I was able to make this work was via the useEffect hook.

Reply:
can you use only the `handlePageChange`? So we can remove the currentPage state

Reply:
I know it seems intuitive but the issue with that is because the same click event of nav/prev and page number buttons also triggers the pagination's state to change. 
So, in order to create some 'delay' to give enough time for `navButtonClicked` to be populated with the label, we need the useffect to be triggered on state change and then pass that new state to the consumers

Reply:
Which function is first called? If it is the `handleNavButtonClick ` then you can also use the `useRef()` to store the data